lib/cc/engine/analyzers/file_list.rb

```diff
@@ -9,7 +11,7 @@ def initialize(engine_config:, default_paths:, language:)
         end
 
         def files
-          matching_files - excluded_files
+          Array(matching_files) & Array(included_files)
```

Comment:
No, we only want files that are in both arrays since `include_paths` is created from the `.codeclimate.yml`s `exclude_paths` and this engine allows you to specify your own paths for each language. So you want to only take the matching files from the total available pool (`include_paths`) and get the files that match from that pool (matching_files here).